trunk/Admin/AdminQuickbarAdmin.php

```diff
@@ -0,0 +1,446 @@
+        add_action( 'elementor/editor/before_enqueue_scripts', [ $this, 'enqueueScripts' ], 99999 );
+
+    }
+
+
+    /**
+     * Set post-types and filtered post-types
+     */
+    public function setPostTypes() {
+        $this->postTypes = get_post_types( [], 'object' );
+
+        $this->filterPostTypes = explode( ',', 'nav_menu_item,revision,custom_css,customize_changeset,'
+            . 'oembed_cache,ocean_modal_window,nxs_qp' );
+
+        foreach ( $this->postTypes as $postType ) {
+            if ( in_array( $postType->name, $this->filterPostTypes ) ) {
+                continue;
+            }
+            $this->filteredPostTypes[] = $postType;
+        }
+    }
+
+    /**
+     * Adds the sidebar to footer
+     *
+     * @param string $data
+     * @return string $data
+     * @throws \ImagickException
+     */
+    public function renderSidebar( $data ) {
+        $this->setPostTypes();
+        $postTypeLoop = $this->getLoopPostTypes();
+        $currentPost = filter_input( INPUT_GET, 'post' );
+
+        $addNewPosts = new Template( self::PartialDir . '/add-new-post.php', [
+            'filteredPostTypes' => $this->filteredPostTypes,
+        ] );
+
+        $tpl = new Template( self::PartialDir . '/admin-quickbar-admin-display.php', [
+            'postTypeLoop' => $postTypeLoop,
+            'currentPost' => $currentPost,
+            'addNewPosts' => $addNewPosts,
+        ] );
+        $tpl->render();
+
+        return $data;
+    }
+
+    /**
+     * Gets rendered post-type loop
+     *
+     * @return string
+     * @throws \ImagickException
+     */
+    public function getLoopPostTypes() {
+        $output = '';
+        foreach ( $this->postTypes as $postType ) {
+            if ( in_array( $postType->name, $this->filterPostTypes ) ) {
+                continue;
+            }
+
+            $posts = $this->getPostsByPostType( $postType );
+            $countPostType = $posts['count'];
+            $cats = $posts['cats'];
+
+
+            if ( empty( $cats ) || empty( $countPostType ) ) {
+                continue;
+            }
+
+            $categoryLoop = $this->getLoopCategories( $postType, $cats );
+
+            $tpl = new Template( self::PartialDir . '/loop-post-types.php', [
+                'postType' => $postType,
+                'categoryLoop' => $categoryLoop,
+            ] );
+            $output .= $tpl->getRendered();
+        }
+
+        return $output;
+    }
+
+    /**
+     * Get rendered category loop
+     *
+     * @param $postType
+     * @param $cats
+     * @return string
+     * @throws \ImagickException
+     */
+    public function getLoopCategories( $postType, $cats ) {
+        $output = '';
+
+        foreach ( $cats as $catName => $posts ) {
+            if ( empty( $posts ) ) {
+                continue;
+            }
+            if ( !$postType->hierarchical ) {
+                $output .= '<div class="admin-quickbar-category">' . $catName . '</div>';
```

Comment:
HTML should only be used in template files.
I suggest to return an array of cateory names and pass it to the `loop-post-types` partial in which handles the rendering.

Reply:
Done